target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-301w.dts

```diff
@@ -0,0 +1,836 @@
+			bias-pull-down;
+		};
+		led_lan1_green {
+			pins = "gpio6";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan1_amber {
+			pins = "gpio7";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};  
+		led_lan2_green {
+			pins = "gpio8";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan2_amber {
+			pins = "gpio11";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan3_green {
+			pins = "gpio12";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan3_amber {
+			pins = "gpio13";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_lan4_green {
+			pins = "gpio14";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_lan4_amber {
+			pins = "gpio15";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_10g_1_green {
+			pins = "gpio51";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_10g_1_amber {
+			pins = "gpio52";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_10g_2_green {
+			pins = "gpio54";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_10g_2_amber {
+			pins = "gpio56";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		}; 
+		led_wifi_green {
+			pins = "gpio42";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+    };
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&prng {
+	status = "okay";
+};
+
+&ssphy_0 {
+	status = "ok";
+};
+
+&qusb_phy_0 {
+	status = "ok";
+};
+
+&ssphy_1 {
+	status = "ok";
+};
+
+&qusb_phy_1 {
+	status = "ok";
+};
+
+&usb_0 {
+	status = "ok";
+};
+
+&usb_1 {
+	status = "ok";
+};
+
+
+&cryptobam {
+	status = "okay";
+};
+
+&crypto {
+	status = "okay";
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qpic_nand {
+	status = "disabled";
+
+};
+
+&blsp1_spi1 { /* BLSP1 QUP1 */
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	cs-select = <0>;
+	status = "ok";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0>;
+		compatible = "n25q128a11";
+		linux,modalias = "m25p80", "n25q128a11";
+		spi-max-frequency = <50000000>;
+		use-default-sizes;
+		m25p,fast-read;
+
+		partitions {
+			compatible = "qcom,smem-part";
+		};
+	};
+};
+
+&mdio {
+	status = "okay";
+
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+	reset-gpios = <&tlmm 37 GPIO_ACTIVE_LOW>;
+
+	ethernet-phy@16 {
+		reg = <16>;
+	};
+
+	ethernet-phy@17 {
+		reg = <17>;
+	};
+
+	ethernet-phy@18 {
+		reg = <18>;
+	};
+
+	ethernet-phy@19 {
+		reg = <19>;
+	};
+
+	ethernet-phy@8 {
+		compatible ="ethernet-phy-ieee802.3-c45";
+		reg = <8>;
+        reset-gpios = <&tlmm 44 GPIO_ACTIVE_LOW>;
+	};
+	
+    ethernet-phy@0 {
+		compatible ="ethernet-phy-ieee802.3-c45";
+		reg = <0>;
+        reset-gpios = <&tlmm 59 GPIO_ACTIVE_LOW>;
```

Comment:
Broken indent

Reply:
Also, can you sort them by reg property

Reply:
both done

Reply:
Only one thing that is not working is the second 10G phy and and don't know why. I'm pretty sure this was working a few days ago but now there is only a physical link and link activity (according to the phy leds) but there is no ip traffic.

I already tried switching the gpio-reset gpio's, removing the reset-gpios...but it's still not working.
Maybe you have an idea.

The first 10G works like a charm.

Reply:
I would bet that it's due to the reset GPIO not actually being toggled fast enough.
You can try adding a reset-post-delay-us property so it waits X amount of time before trying to probe after reset is de-asserted.
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Documentation/devicetree/bindings/net/mdio.yaml?h=v5.15.1#n42

I had the same issue on AX9000 where the QCA8081 wasn't replying fast enough for auto-ID, but I then set the PHY ID directly in the compatible and it works that way.
https://github.com/robimarko/openwrt/blob/IPQ807x-5.10-backports/target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax9000.dts#L227

Reply:
This is really strange...the second 10G port works only as wan (uplink) port.

I can't get them to work as LAN-Port.

I tried both of your solutions, without any luck.

So I changed the lan/wan port definition and changed the commit messages regarding the default network configuration.

Reply:
I think I know whats the issue, it's the crap QCA switch driver.
As they are not using DSA they are using the following DTS properties:
```
switch_lan_bmp = <0x3e>; /* lan port bitmap */
switch_wan_bmp = <0xc0>; /* wan port bitmap */
```

According to these then SSDK will configure the forwarding register and allow/disallow forwarding between certain ports.
Usually the WAN port can only forward to the CPU port, hence why you cant use it as LAN

Reply:
Thanks for the explanation, so this means is is hardcoded in the dts.